sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -418,4 +430,67 @@ public Map.Entry<String, String> next() {
             return new AbstractMap.SimpleImmutableEntry<>(key, value);
         }
     }
+
+    public static Context enableSyncRestProxy(Context context) {
+        if (context == null || context == Context.NONE) {
+            return STATIC_ENABLE_REST_PROXY_CONTEXT;
+        } else {
+            return context.addData(HTTP_REST_PROXY_SYNC_PROXY_ENABLE, true);
+        }
+    }
+
+    public static  <T> Response<T> executeOperation(Supplier<Response<T>> operation, Duration timeout) {
+        try {
+            return timeout != null
+                ? THREAD_POOL.submit(() -> operation.get()).get(timeout.toMillis(), TimeUnit.MILLISECONDS)
+                : operation.get();
+        } catch (ExecutionException | TimeoutException | InterruptedException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    public static ExecutorService getThreadPoolWithShutdownHook() {
+        ExecutorService threadPool = Executors.newCachedThreadPool();
+        registerShutdownHook(threadPool);
+        return threadPool;
+    }
+
+    public static void registerShutdownHook(ExecutorService threadPool) {
+        long halfTimeout = TimeUnit.SECONDS.toNanos(THREADPOOL_SHUTDOWN_HOOK_TIMEOUT_SECONDS) / 2;
+        Thread hook = new Thread(() -> {
+            try {
+                threadPool.shutdown();
+                if (!threadPool.awaitTermination(halfTimeout, TimeUnit.NANOSECONDS)) {
+                    threadPool.shutdownNow();
+                    threadPool.awaitTermination(halfTimeout, TimeUnit.NANOSECONDS);
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                threadPool.shutdown();
+            }
+        });
+        Runtime.getRuntime().addShutdownHook(hook);
+    }
+
+    public static String tagsToString(Map<String, String> tags) {
```

Comment:
Was this copied from elsewhere? This seems familiar. And why was this added?

Reply:
This was copied from async client I believe when adding sync code to sync client.
Updated async client to reference it from here.
We can move it back to clients too and have a static package private copy there, if its not needed here.